### PR TITLE
Improve tests and allow BigInts on `findByPk`

### PR DESCRIPTION
**Pull request by papb**
Hello! :smiley: 

This PR does several things:

* Refactors some tests using async/await instead of promises (and other readability improvements)
* Fixes some tests that weren't waiting for promises correctly and passing without really testing anything ([example](https://github.com/cockroachdb/sequelize-cockroachdb/blob/a230506c544beb6ef1163552a817bab877daf565/tests/int_test.js#L30-L34))
* [Adds a thorough cleanup step after each test](https://github.com/cockroachdb/sequelize-cockroachdb/pull/65/files#diff-1bc97b15b1e837743f590899c1c0fd36eab5a1bf21df692490899220dca7b160R40-R60) to avoid interference between tests - this also allowed me to unskip an entire `describe` suite
* [Adds a "basic usage" test in which basic Sequelize things are used](https://github.com/papb/sequelize-cockroachdb/runs/2042599563#step:7:12) - I suggest using this test to back the claim of "alpha support" :tada:
  * When I added this test, it failed, but the solution was simple (just allow BigInts on `findByPk`), so [I did it myself](https://github.com/cockroachdb/sequelize-cockroachdb/pull/65/files#diff-88a99604221821c7258fe9c24fc58a4ef42242b97aeae97d6c438db3f62be6cd)! :tada:
* Enables skipping the huge & slow sequelize integration tests by writing the following in the *end* of the commit message:
  > `[[skip sequelize integration tests]]`
* Converts the `see-upsert-working.js` file (that [I created](https://github.com/cockroachdb/sequelize-cockroachdb/pull/47/files#diff-30b1e71331ebe473478909f9ce8222a6bf60ce89906324d660e64bdd083c077e) in #47) into a real test

----

You can see the new `[[skip sequelize integration tests]]` feature [working at my fork](https://github.com/papb/sequelize-cockroachdb/runs/2042599540) (once this PR is merged, it will start being applied on this repo automatically) - it only has 2 checks instead of over 200. This is great for faster feedback and for saving energy, for situations in which we don't really want to see all those checks (which is exactly the case for this PR, since I didn't change anything that I expect to make a difference on those results).

----

Suggestion: I think an admin of this repository should set the two "main" checks as required for merges (at some point, perhaps not now while many changes are still being done):

* Main - CRDB v20.2.4, Sequelize v5
* Main - CRDB v20.2.4, Sequelize v6

This can be done in the *Require status checks to pass before merging* section, [here](https://github.com/cockroachdb/sequelize-cockroachdb/settings/branches):

image

### CODE REVIEW

**Comment by vy-ton** [comment]
@RichardJCai Pedro (@papb) was the Sequelize maintainer we were first working with. This is his final PR with some improvements that he shared. Can you help merge this work with the ongoing work from our other contractors?

**Comment by RichardJCai** [comment]
> @RichardJCai Pedro (@papb) was the Sequelize maintainer we were first working with. This is his final PR with some improvements that he shared. Can you help merge this work with the ongoing work from our other contractors?

Ack, will do.

File: tests/drop_enum_test.js [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
-        },
-        sequence: {
-          type: DataTypes.INTEGER,
-          allowNull: true
-        },
-        name: {
-          type: DataTypes.STRING,
-          allowNull: true
-        }
+      await this.queryInterface.createTable('menus', {
```
Comment (RichardJCai) [inline tests/drop_enum_test.js:28]: Just curious, what's the reason behind this change? I use the sequelize equivalent tests for reference and they define the params the way we did before 

https://github.com/sequelize/sequelize/blob/main/test/integration/query-interface/dropEnum.test.js#L21

File: tests/helper.js [hunk 2]
```diff
@@ -36,6 +37,24 @@ before(function() {
   });
 });
 
-after(function() {
-  this.sequelize.close();
-})
+afterEach(async function() {
+  // https://github.com/sequelize/sequelize/blob/29901187d9560e7d51ae1f9b5f411cf0c5d8994a/test/support.js#L136
```
Comment (RichardJCai) [inline tests/helper.js:41]: Can we pull this logic into a helper function as well?

**Review by RichardJCai — COMMENTED** [review]
Thanks this is great! I just have some minor comments

File: tests/drop_enum_test.js (hunk 1 above)
Comment (papb) [reply tests/drop_enum_test.js:28]: Oh! This change is not important, it was just a refactor that I felt like doing. The behavior is unchanged, since `allowNull` is already `true` by default. I prefer shorter tests, so I like to omit defaults like these. However, I didn't notice this was based on an existing Sequelize test. Thanks for bringing it to my attention, I've just opened [a PR](https://github.com/sequelize/sequelize/pull/13084) on Sequelize itself to perform the same refactor there.

File: tests/helper.js (hunk 2 above)
Comment (papb) [reply tests/helper.js]: Sure! Like this? 7a1bd15

Comment (RichardJCai) [reply tests/helper.js]: I prefer `cleanupDatabase` over `cleanDatabase`

**Review by RichardJCai — APPROVED** [review]
Thanks! Just one more nit comment about the helper function naming.

**Comment by papb** [comment]
@RichardJCai That's better indeed. 5e41170

File: tests/helper.js (hunk 2 above)
Comment (papb) [reply tests/helper.js]: That's better indeed. 5e41170

**Comment by RichardJCai** [comment]
LGTM thanks!

**Review by RichardJCai — APPROVED** [review]
